onnxscript/rewriter/ort_fusions/attention_test.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+from __future__ import annotations
+
+import unittest
+
+import numpy as np
+import packaging.version
+import parameterized
+
+import onnxscript
+import onnxscript.ir as ir
+import onnxscript.rewriter.ort_fusions._core as xformers
+from onnxscript import FLOAT, script
+from onnxscript import opset18 as op
+from onnxscript.ir.passes.common import shape_inference
+from onnxscript.rewriter.ort_fusions._test_utils import ORT_VERSION, assert_allclose, ort_run
+
+msft_op = onnxscript.values.Opset("com.microsoft", 1)
+
+
+class TestAttentionFusion(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.batchsize = 2
+        self.seqlen = 8
+        self.max_seqlen = 32
+        self.headsize = 16
+        self.num_heads = 10
+        self.input_hidden_size = self.headsize * self.num_heads
+        self.q_hidden_size = 160
+        self.k_hidden_size = 160
+        self.v_hidden_size = 160
+
+    def random_inputs(self, with_past=False):
+        """Generate random inputs for the model."""
+        B = self.batchsize
+        S = self.seqlen
+        M = self.max_seqlen
```

Comment:
I guess "max_seqlen" is a bit different from "total_seqlen"? Does it impact the fusion or the test below which of the two we use?

Reply:
I mean, do we need to separately test the two cases where the buffer is (a) equal-to and (b) greater-than what is needed for this run? Or, does it not matter?

Reply:
That is correct, max_seq_len is different from total_seq_len. However max_seq_len is only used when past and present share the same buffer, the op's parameter `past_present_share_buffer` is set to 1 (it is zero by default) and `past_sequence_length` is provided as an input. I guess this PR doesn't cover this pattern so we dont need to test this here. I changed past shape sequence length from total_seq_len to past_seq_len to reflect this.